src/pybind/mgr/cephadm/module.py

```diff
@@ -1250,13 +1258,19 @@ def _create_osd(self, all_hosts_, drive_group):
                 if len(list(set(devices) & set(osd['devices']))) == 0 and osd.get('lv_path') not in devices:
                     self.log.debug('mismatched devices, skipping %s' % osd)
                     continue
+                if osd_id not in osd_uuid_map:
+                    self.log.debug('osd id %d does not exist in cluster' % osd_id)
+                    continue
+                if osd_uuid_map[str(osd_id)] != osd['tags']['ceph.osd_fsid']:
```

Comment:
`osd_uuid_map` is of type `Dict[str, str]`? 

And according to 

https://github.com/ceph/ceph/blob/afc28b96de738dbca674e5d7c34660f1de2b1ab4/src/ceph-volume/ceph_volume/devices/lvm/listing.py#L243

`osd_id` is already a `str`. can we simply drop `str()` here? Right now this looks as if `osd_id` is an integer.